scripts/valhalla_build_config

```diff
@@ -534,54 +535,65 @@ help_text = {
   }
 }
 
-#returns a list of leaves of the tree, `\0` separated, stops at non dicts
-#while doing so it also adds arguments to the parser
-def add_leaf_args(path, tree, leaves, parser):
-  #if we are at a dict go deeper
-  if isinstance(tree, dict):
-    for k in tree:
-      v = tree[k]
-      add_leaf_args('\0'.join([path, k]) if len(path) else k, v, leaves, parser)
-  #we've reached a leaf
-  else:
-    keys = path.split('\0')
-    h = help_text
-    for k in keys:
-      h = h[k]
```

Comment:
really hard to see what's going on in this diff.. I hardly changed the logic

- I just moved the comments above the functions into their docstrings
- added the type hints
- renamed some variables to not shadow those from the outer scope (even though it's exactly those which they're referencing, I find it less confusing)